pyretailscience/plots/period_on_period.py

```diff
@@ -0,0 +1,143 @@
+    (0, (6, 6)),  # loosely dashed
+    (0, (3, 5, 1, 5)),  # loosely dashdotted
+    (0, (3, 5, 1, 5, 1, 5)),  # loosely dashdotdotted
+]
+
+
+def plot(
+    df: pd.DataFrame,
+    x_col: str,
+    value_col: str,
+    periods: list[tuple[str | datetime, str | datetime]],
+    x_label: str | None = None,
+    y_label: str | None = None,
+    title: str | None = None,
+    source_text: str | None = None,
+    legend_title: str | None = None,
+    move_legend_outside: bool = False,
+    ax: Axes | None = None,
+    **kwargs: dict[str, any],
+) -> Axes:
+    """Plot multiple overlapping periods from a single time series as individual lines.
+
+    This function is used to align and overlay several time intervals from the same
+    dataset to facilitate visual comparison. Each period is realigned to the reference
+    start date and plotted as a separate line using a distinct linestyle.
+
+    Note:
+        The `periods` argument accepts a list of (start_date, end_date) tuples,
+        which define the time windows to overlay. Each element in the tuple can be either
+        a string (e.g., "2022-01-01") or a `datetime` object. You can use
+        `find_overlapping_periods` from `pyretailscience.utils.date` to generate
+        the `periods` input automatically.
+
+    Args:
+        df (pd.DataFrame): Input DataFrame containing the time series data.
+        x_col (str): Name of the column representing datetime values.
+        value_col (str): Name of the column representing the y-axis values (e.g. sales, counts).
+        periods (List[Tuple[Union[str, datetime], Union[str, datetime]]]):
+            A list of (start_date, end_date) tuples representing the periods to plot.
+        x_label (Optional[str]): Custom label for the x-axis.
+        y_label (Optional[str]): Custom label for the y-axis.
+        title (Optional[str]): Title for the plot.
+        source_text (Optional[str]): Text to show below the plot as a data source.
+        legend_title (Optional[str]): Title for the plot legend.
+        move_legend_outside (bool): Whether to place the legend outside the plot area.
+        ax (Optional[Axes]): Matplotlib Axes object to draw on. If None, a new one is created.
+        **kwargs: Additional keyword arguments passed to the base line plot function.
+
+    Returns:
+        matplotlib.axes.Axes: The matplotlib Axes object with the completed plot.
+
+    Raises:
+        ValueError: If `periods` is an empty list.
+    """
+    if not periods:
+        raise ValueError("The 'periods' list must contain at least one (start, end) tuple.")
+
+    periods = [(pd.to_datetime(start), pd.to_datetime(end)) for start, end in periods]
+    start_ref = periods[0][0]
+
+    sorted_periods = sorted(periods, reverse=True, key=lambda x: pd.to_datetime(x[0]))
+
+    ax = ax or plt.gca()
+
+    period_styles = {}
+    for idx, period in enumerate(sorted_periods):
+        period_styles[period] = LINE_STYLES[idx % len(LINE_STYLES)]
```

Comment:
Very minor but could we implement this using this syntax please
```suggestion
    period_styles = { period: LINE_STYLES[idx % len(LINE_STYLES)]
    for idx, period in enumerate(sorted_periods)}
```